spec/hydra/works/models/concerns/generic_file/characterization_spec.rb

```diff
@@ -0,0 +1,157 @@
+require 'spec_helper'
+
+describe Hydra::Works::GenericFile::Characterization do
+  # Demonstration class stand in for what downstream applications implement.
+  let(:demoClass)  do
+    class DemoClass < Hydra::Works::GenericFile::Base
+      include Hydra::Works::GenericFile::Characterization
+    end
+  end
+
+  # The is the only test context that will not mock out Characterization.characterize
+  #   Remove after development as this is more of an integration test
+  context "using fits.sh via Hydra::FileCharacterization" do
+
+    let(:filename)     { 'sample-file.pdf' }
+    let(:mimetype)     { 'application/pdf' }
+    let(:file_content) { File.open(File.join(fixture_path, filename)) }
+    let(:file)         { Hydra::PCDM::File.new.tap{ |f| f.content = file_content } }
+    let(:type)         { ::RDF::URI("http://pcdm.org/use#OriginalFile") }
+
+    subject { demoClass.new(id:'demo/01/01') }
+
+    before do
```

Comment:
@terrellt I was mocking out the add file to generic file service so that generic files did not have to be saved in order to have a file attached as content.  This is what I came up with.  My next plan is to try to extract and encapsulate the logic in a spec_helper function, but have not had luck doing that so far.